### PR TITLE
Delaney, kaggle and nci example bugfixes and ChEMBL dataset

### PR DESCRIPTION
**Pull request by CCXD**
The loaders do no longer have the num_shards_per_batch argument, and would result in unexpected argument error.

Delaney mol graph data has 71 mol features, not 75.

NCI dataset loader already handled the random splitting before returning the datasets. No longer required to do it manually.

ChEMBL database is from https://www.ebi.ac.uk/chembl/ v22.1
The records in this dataset meet some quality requirements: From literature, tested on known protein target, no functional assays.

I have a larger dataset of 430 000 records, but wanted to see if this set was worth benchmarking on, as it is very sparse. Maybe compounds that have activities on many targets can be selected?

Number of compounds in train set
15284
Number of compounds in validation set
1911
Number of compounds in test set
1911
Number of tasks (protein targets)
92

Prelim results:
Deepchem MTQSAR
Train scores
{'mean-pearson_r2_score': 0.45722603229151743}
Validation scores
{'mean-pearson_r2_score': 0.43894487056784254}
Test scores
{'mean-pearson_r2_score': 0.40626273298153748}

Deepchem Graph Conv
Train scores
{'mean-pearson_r2_score': 0.10596338034798811}
Validation scores
{'mean-pearson_r2_score': 0.28837691403511423}

### CODE REVIEW

**Comment by rbharath** [comment]
@CCXD Thanks for contributing this! Adding Chembl has been on our wishlist for some time, but we haven't been able to get to it. Couple of comments:

- Would you mind gzipping chembl.csv? Will help save on space on the repo.
- I believe `n_feat` should be `75` for both `delaney_graph_conv.py` and `chembl_graph_conv.py`. The latest changes in #347 changed graph-convs to support decavalent compounds, so we need to change `n_feat` to `75` throughout. @miaecle Is my summary correct?
- In `nci_rf.py`, thanks for getting rid of the unneeded splitters. Would you mind adding in a line `(train_dataset, valid_dataset, test_dataset) = nci_dataset`?

None of these are major issues, and I'm actually OK with merging as-is (since all changes are in `examples/` which isn't part of test-suite). I have a large refactoring change to the graph-conv coming up, and I'd like to get this one in before then.

**Comment by miaecle** [comment]
@CCXD Thanks a lot for making these changes, very glad to see adding new datasets into the collection!
@rbharath Yes, the only significant change due to graph-conv modification is setting `n_feat` to 75.

**Comment by CCXD** [comment]
@rbharath Applied those fixes, and changed the chembl set to 691 tasks.
I added a sparse set, where most of the compounds have an activity value for one target.
The "5thresh" set is a smaller set, where all compounds have activity values for at least 5 targets.
In addition to this, I added a split option "year" for this dataset. As the data is from literature, it can be divided into publication dates. All the data that was published in and before the year 2013 is in the training and validation set, and test set is 2014 and after. 

5thresh 
train: 14620
test:  5252
valid: 4000

sparse
train: 137594
test:  66652
valid: 20000

**Comment by rbharath** [comment]
@CCXD Thanks for the contribution! I think everything looks good, so going to go ahead and merge. Would be great to see more DeepChem PR's coming our way :-)

@miaecle Thanks for confirming the fingerprint size! When you have some free time, would you mind integrating this dataset into the benchmarking suite?

**Comment by miaecle** [comment]
@rbharath Sure, I will be very glad to merge it!

**Comment by Moooooooon** [comment]
> ChEMBL database is from https://www.ebi.ac.uk/chembl/ v22.1
The records in this dataset meet some quality requirements: From literature, tested on known protein target, no functional assays.

@CCXD 
Thanks for contributing Chembl dataset!
I would like to ask you a simple question about your Chembl data.

- In the case of 5thresh, there is a protein in the column and a drug in the row. What does the number at the contact of the column and row mean?

I tried to find the meaning on the official site or official database, but I could not find it.
Thank you.

**Comment by CCXD** [comment]
@Moooooooon The number is the pChEMBL value, the -log activity of that compound on that specific protein. This activity can be IC50, Ki, Kd, but functional assays are removed so no EC50. 

https://www.ebi.ac.uk/chembl/faq#faq67

**Comment by Moooooooon** [comment]
@CCXD Thank you for the detailed explanation!
